crates/gitbutler-tauri/src/projects.rs

```diff
@@ -57,6 +58,23 @@ pub mod commands {
         handle.state::<Controller>().list().map_err(Into::into)
     }
 
+    /// This trigger is the GUI telling us that the project with `id` is now displayed.
+    ///
+    /// We use it to start watching for filesystem events.
+    #[tauri::command(async)]
+    #[instrument(skip(handle), err(Debug))]
+    pub async fn set_project_active(handle: tauri::AppHandle, id: &str) -> Result<(), Error> {
+        let id: ProjectId = id.parse().context(error::Context::new_static(
+            Code::Validation,
+            "Malformed project id",
+        ))?;
```

Comment:
@mtsgrd didn't we end up fixing the project ID acquisition stuff? Or did that never end up landing?

Reply:
We didn't change anything, we just dropped the idea of custom validation failure responses since they are not actually handled by the front end. Instead we can rely on the built-in conversion to respond with appropriate failures (including a description of why the param conversion failed).

A different thing here, where could use `Code::Validation` is if e.g. a project id does not exist.. but we're not there yet.

Reply:
Yeah let's do that. We want to move to that app-wide anyway, I could have sworn we had done that a while back. Maybe some regressions seeped in from somewhere, not sure.

Reply:
Alright, https://github.com/gitbutlerapp/gitbutler/pull/3471/commits/62b1c49372972139feec9750a198da8802db79fb should bring it back to the desired state.